SWI-cpp2.cpp

```diff
@@ -858,7 +858,7 @@ void PlWrapDebug(const char*msg) {
 		 *******************************/
 
 _SWI_CPP2_CPP_inline
-PlStream::PlStream(PlTerm& stream, int flags)
+PlStream::PlStream(const PlTerm& stream, int flags)
```

Comment:
The `const` is not really true.   The PL_get_stream() locks the stream.

Reply:
Welcome to the world of `const` in C++, where  `const` doesn't mean immutable. :)
More precisely, the `const PlTerm& stream` means that the contents of `stream` (that is, the field `IOSTREAM *s_`) can't be changed, but allows changing whatever `stream.s_` points to.

I'll change this to the following, which is possibly a tiny bit more efficient:
```
PlStream::PlStream(PlTerm stream, int flags)
```
which copies the `PlTerm`, which simply contains a `term_t`, which the `unwrap()` gets. 

Without the `const`, I  get this (which confuses and frightens me):
```
In file included from ../packages/cpp/test_cpp.cpp:63:
../packages/cpp/test_cpp.cpp: In function ‘foreign_t pl_portray_my_blob__2(PlTermv)’:
../packages/cpp/SWI-cpp2.h:1591:23: error: cannot bind non-const lvalue reference of type ‘PlTerm&’ to an rvalue of type ‘PlTerm’
 1591 | #define PL_A1  PL_av[0]
      |                ~~~~~~~^
../packages/cpp/SWI-cpp2.h:1603:12: note: in expansion of macro ‘PL_A1’
 1603 | #define A1 PL_A1
      |            ^~~~~
../packages/cpp/test_cpp.cpp:1429:17: note: in expansion of macro ‘A1’
 1429 |   PlStream strm(A1, 0);
      |                 ^~
In file included from ../packages/cpp/SWI-cpp2.h:1864,
                 from ../packages/cpp/test_cpp.cpp:63:
../packages/cpp/SWI-cpp2.cpp:861:28: note:   initializing argument 1 of ‘PlStream::PlStream(PlTerm&, int)’
  861 | PlStream::PlStream(PlTerm& stream, int flags)
```